examples/cmd/nsc/nsc.go

```diff
@@ -46,42 +47,46 @@ func main() {
 		workspace, err = nsmd.RequestWorkspace()
 		if err != nil {
 			logrus.Fatalf("nsc: failed set up client connection, error: %+v, exiting...", err)
-			os.Exit(1)
 		}
 	}
 
 	clientSocket := path.Join(workspace, nsmd.ClientSocket)
 
 	logrus.Infof("Connecting to nsm server on socket: %s...", clientSocket)
 	if _, err := os.Stat(clientSocket); err != nil {
-		logrus.Errorf("nsc: failure to access nsm socket at %s with error: %+v, exiting...", clientSocket, err)
-		os.Exit(1)
+		logrus.Fatalln("nsc: failure to access nsm socket at %s with error: %+v, exiting...", clientSocket, err)
 	}
 
 	conn, err := tools.SocketOperationCheck(clientSocket)
 	if err != nil {
 		logrus.Fatalf("nsm client: failure to communicate with the socket %s with error: %+v", clientSocket, err)
-		os.Exit(1)
 	}
 	defer conn.Close()
 
 	// Init related activities start here
-	nsmConnectionClient := nsmconnect.NewClientConnectionClient(conn)
+	nsmConnectionClient := networkservice.NewNetworkServiceClient(conn)
 
-	_, err = nsmConnectionClient.RequestConnection(context.Background(), &nsmconnect.ConnectionRequest{
-		RequestId:          linuxNS,
-		LinuxNamespace:     linuxNS,
-		NetworkServiceName: "gold-network",
-		LocalMechanisms: []*common.LocalMechanism{
-			&common.LocalMechanism{
-				Type: common.LocalMechanismType_KERNEL_INTERFACE,
+	_, err = nsmConnectionClient.Request(context.Background(), &networkservice.NetworkServiceRequest{
+		Connection: &networkservice.Connection{
+			ConnectionId:   netns + "-" + strconv.FormatInt(time.Now().Unix(), 36),
+			NetworkService: "icmp-responder",
+			ConnectionContext: &networkservice.ConnectionContext{
+				ConnectionContext: map[string]string{
+					"requires": "src_ip,dst_ip",
+				},
+			},
+			Labels: make(map[string]string),
+		},
+		LocalMechanismPreference: []*common.LocalMechanism{
```

Comment:
Does NSM always choose the first LocalMechanism from this array?

Reply:
NSM treats this as a preference ordered list.  So it picks the first one it can *do*.  Given where we are at the moment, just pick the first one.

Reply:
So NSC gives to NSM a preference ordered list and NSE gets mechanisms from NSM and chooses itself? Is it correct understanding?
I thought you discussed in IRC some kind of symmetry, so I'm little confused :)

Reply:
Yes :)

The symmetry here is that the NetworkService API that the NSC uses to address the NSM is the same NetworkService API the NSM uses to address the NSE.